### ISSUE TITLE
Merge Latest uGT Emulation (included bitwise correlation conditions) into CMSSW

### ISSUE THREAD

**Issue by mulhearn** [cms-l1t-offline/cmssw#261]
Here we will track the steps required to get the latest uGT Emulation merged into CMSSW 80x and 81x.

1) Rebase global development branch onto CMSSW_8_0_5 + PR cms-sw#14141 (DONE, see this branch:  https://github.com/cms-l1t-offline/cmssw/tree/l1t-global-CMSSW_8_0_5)

Brian and Mike both find bitwise equivalent results with their respective previous versions, with caveat that UTM library is not updated in CMSSW_8_0_5, so Brian manually updated for his test. 

2) Cleanup CondFormats and make a separate PR with only new Conditions to limit sign-off burden on main PR. cms-sw#14201 and cms-sw#14202.  (DONE)

3) Get updated UTM library into CMSSW (Takashi).  Note we can continue to integrate everything else in parallel, but we won't have correlation conditions until the new UTM library is in place.

4) Update HLT Seed Filter to work with updated GT output (DONE)

5)  1t-tsg-v5 version of the GT Emulator uses GlobalStableParameter which is set via StableParametersTrivialProducer.  The updated version uses L1TGlobalParameters via helper class GlobalParamsHelper and ES Producer L1TGlobalParamsESProducer.  This will require some updates to HLT configs in addition to our own.  Note that the old condition and producer are kept around, so they can be harmless set as before if more convenient.

Andrea suggested a nice approach to handling this, as explained here:  cms-sw#14207
But rejected by Martin!  Then resolved by Mike.  (DONE)

6) Make Main PR into CMSSW.  81X version is here: cms-sw#14282 (merged)
80X version is here: cms-sw#14290

7) We had some RelVal failures in 81X due to this PR, which were addressed with bugfixes:  cms-sw#14292, cms-sw#14296, cms-sw#14297.  The commits from these PRs have been appended directly to the 80x PR.

**Comment by tmatsush** [comment]
3) for 80X, already requested  as https://github.com/cms-sw/cmsdist/pull/2254
    for 81X, needs a test environment with Xerces-3.1

**Comment by mulhearn** [comment]
Oooof, just spend a long night wrestling with RelVal failures.  1st round was CondFormats.  2nd round was DataFormats... waiting for test to finish to see if there is 3rd round.

**Comment by mulhearn** [comment]
If these latest RelVal tests pass, I think I am maybe down to just adding a new HLT fragment for the new global stable parameters. (Item 5)

**Comment by mulhearn** [comment]
These were the problematic ones:

```
1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15INPUT+DIGIUP15+RECOUP15+HARVESTUP15 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Fri Apr 22 06:39:48 2016-date Fri Apr 22 06:27:57 2016; exit: 0 0 0 0
1330.0_ZMM_13+ZMM_13INPUT+DIGIUP15+RECOUP15+HARVESTUP15 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Fri Apr 22 06:40:13 2016-date Fri Apr 22 06:27:58 2016; exit: 0 0 0 0
25202.0_TTbar_13+TTbar_13INPUT+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25 Step0-PASSED Step1-PASSED Step2-PASSED Step3-PASSED  - time date Fri Apr 22 07:01:03 2016-date Fri Apr 22 06:28:02 2016; exit: 0 0 0 0
3 3 3 3 tests passed, 0 0 0 0 failed
```

**Comment by mulhearn** [comment]
Resolved the confusion on the externals requests for UTM... there was some confusion about an update to XERCES version.  Since we are happy to leave XERCES version at 2.8 the same external can go into 80x and 81x.  David is on that one.

**Comment by mulhearn** [comment]
The complete set of problematic RelVals in the 81x IB (not all of them due to us) were:
- 135.11,135.3,135.7,1307,1349,25403,25408,250199,250399,250403,250409
- 1303,1313,1321,1322,1336,1341,1343,1345,1360,250202,250206 

After testing the 81x bug fixes, the only failures I have upon rerunning appear unrelated to L1, in RelVals:
1360.0, 250403.0, 25403.0, and 135.7.

**Comment by mulhearn** [comment]
@puigh @blwiner please have a look at the update PR for 80x cms-sw#14290 and make certain it behaves as you expect in your own tests.  We had several IB failures due to (1) me inadvertently blowing away a bug I already fixed once, (2) crashes in the Correlation conditions that were due to the HW ET being larger than the scale.  I made fixes in the last four commits on the branch, or PRs cms-sw#14292, cms-sw#14296, and cms-sw#14297.

I understand (1) but I'm not sure if (2) possibly hides a deeper problem.  For a quick fix I simply adjusted the et to the maximum of the scale in these cases.  We only saw crashes due to jets and sums, but I put protection for all objects.  Please have a closer look at that PR on top of CMSSW_8_0_6.  Note that the test uses the standard sequences and the menu in the GT, which at the moment is L1Menu_Collisions2015_25nsStage1_v7_uGT_v4.

**Comment by mulhearn** [comment]
The 80x version of the PR is merged, and in IB CMSSW_8_0_X_2016-05-02-1100 all RelVals pased (except for 1 unrelated DAS ERROR.)  This one is closed.

**Comment by fwyzard** [comment]
@mulhearn did you get any feedback about 

> I understand (1) but I'm not sure if (2) possibly hides a deeper problem.

?
